tip-wlan/charts/opensync-gw-cloud/templates/deployment.yaml

```diff
@@ -142,7 +153,7 @@ spec:
             - name: tip_wlan_maxHttpThreads
               value: "{{ .Values.scalability.tip_wlan_maxHttpThreads }}"
             - name: JVM_MEM_OPTIONS
-              value: "{{ .Values.scalability.JVM_MEM_OPTIONS }}"
+              value: "{{ .Values.scalability.JVM_MEM_OPTIONS }} -javaagent:/jmx-prometheus-exporter-dir/jmx_prometheus_javaagent-0.14.0.jar=9404:/app/jmx-prometheus-config.yml"
```

Comment:
what about security for the jmx_prometheus_javaagent agent? for both access control and transport (to protect against traffic sniffing/injecting). can we configure TLS with certificate-based auth?

Reply:
Not sure if I understand your issue here. When running the exporter as a Java agent there is no network communication at all. Everything goes through altering/reading bytecode within the JVM. Let me know if I miss something here.

Reply:
So, the agent will not be accepting network connections on port 9404?
I mean connections from things like JMXConsole, which can access any MBean and call methods on them that not only can read things, but to also change the limits on pool sizes, trigger garbage collection, of invoke arbitrary methods on arbitrary java classes.
Maybe I'm missing something and the jmx_prometheus_javaagent does not accept network connections from the outside world? Does it push the data to the prometheus instead?

Reply:
Try connecting jconsole to port 9404. I'm curious what happens.
https://docs.oracle.com/javase/8/docs/technotes/guides/management/jconsole.html

Reply:
https://github.com/prometheus/jmx_exporter 
"This exporter is intended to be run as a Java Agent, exposing a HTTP server and serving metrics of the local JVM. "
To me this feels weird - to run an unprotected HTTP server that is directly embedded as an agent into the bowels of the server process. Even if it does not expose full capabilities of JMX, it's just asking for trouble.

Reply:
The port 9404 is for the HTTP server so a JMX console will not be able to connect to it.

IMO you are a little overcautious here. We are talking about a read only HTTP interface with non confidential (at least for now) information that is never being exposed outside of the cluster.

Reply:
Just to add to the discussion here - read-only HTTP endpoints are pretty common in the Kubernetes-monitoring space. Prometheus scraping works by periodically pulling HTTP endpoints exposed by the application for metrics reporting purposes. As Prometheus' security model suggests (https://prometheus.io/docs/operating/security/) - metrics aren't considered secrets, and despite the fact that Prometheus supports TLS for scraping, many available community exporters rarely implement HTTPs support.

I suggest we go forward with the implementation, and if needed, add the TLS support in the future since the `jmx_exporter` actually supports that.

Reply:
My personal opinion after reading the page quoted (https://prometheus.io/docs/operating/security/) is that prometheus security model is a joke. I would strongly suggest to use TLS wherever it is available. Nobody knows what turns the project takes in the future, nobody knows if at some point the deployment takes a path of having several clusters on public untrusted cloud - for variour reasions - money, politics, scalability requirements, etc. If you select a path of least resistance now (unencrypted internal traffic) - you may put the project into a corner in the future.

Reply:
I don't say that mTLS is a no-go. But it shouldn't be the default and available as an addon for setups that require increased safety measures. It adds complexity that isn't necessary for dev envs and keeps users from getting started with CloudSDK.

Reply:
That decision should be made by @cbuschfb and @Dorongiv.

Reply:
If we put on hold my comments about mTLS/TLS, how far away are we from making this prototype a reality?
This feature can always be configured on/off, and it provides a lot of value for the scalability testing.

Reply:
It's not too big of a deal. The configuration will be the same for all services + some templating to reduce duplications.